ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceBlobSidecarsAvailabilityChecker.java

```diff
@@ -347,8 +347,8 @@ private BlobSidecarsAndValidationResult computeFinalValidationResult(
         Collections.unmodifiableList(completeValidatedBlobSidecars));
   }
 
-  private boolean isBlockOutsideDataAvailabilityWindow() {
-    return !spec.isAvailabilityOfBlobSidecarsRequiredAtSlot(
+  private boolean isStorageOfBlobSidecarsRequiredAtSlot() {
```

Comment:
We should go back to`isBlockOutsideDataAvailabilityWindow` (note that this is negative of `isStorageOfBlobSidecarsRequiredAtSlot`)

Reply:
The only place I see we can use the new Storage param is in the BlobSidecarPruner

Reply:
The `validateImmediately` flow is used for the historical backfill. But I'd do that in the decoupling backfill PR, otherwise we make backfill of blocks stall if the value is used and set too big.

But IMO that method name should be renamed to something more tight to the historical batch sync, otherwise it won't be clear (and risky) that we want to use the storage limit in that case.